GCC 7 - make check fails

### Problem
`make check` return with error
```
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
```

### Operating system, kernel, GCC
```
❯ cat /etc/os-release
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
ID_LIKE=archlinux
ANSI_COLOR="0;36"
HOME_URL="https://www.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"

❯ uname -r           
4.13.3-1-zen

❯ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/7.2.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --enable-libmpx --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --disable-multilib --disable-werror --enable-checking=release --enable-default-pie --enable-default-ssp
Thread model: posix
gcc version 7.2.0 (GCC)
```

### Output
```
❯ make check
Making check in .
make[1]: Wejście do katalogu '/home/adrian/Downloads/protobuf-3.4.1'
make  check-local
make[2]: Wejście do katalogu '/home/adrian/Downloads/protobuf-3.4.1'
Making lib/libgmock.a lib/libgmock_main.a in gmock
make[3]: Wejście do katalogu '/home/adrian/Downloads/protobuf-3.4.1/gmock'
make[3]: 'lib/libgmock.la' jest aktualne.
make[3]: 'lib/libgmock_main.la' jest aktualne.
make[3]: Opuszczenie katalogu '/home/adrian/Downloads/protobuf-3.4.1/gmock'
make[3]: Wejście do katalogu '/home/adrian/Downloads/protobuf-3.4.1/gmock/gtest'
make[3]: 'lib/libgtest.la' jest aktualne.
make[3]: 'lib/libgtest_main.la' jest aktualne.
make[3]: Opuszczenie katalogu '/home/adrian/Downloads/protobuf-3.4.1/gmock/gtest'
make[2]: Opuszczenie katalogu '/home/adrian/Downloads/protobuf-3.4.1'
make[1]: Opuszczenie katalogu '/home/adrian/Downloads/protobuf-3.4.1'
Making check in src
make[1]: Wejście do katalogu '/home/adrian/Downloads/protobuf-3.4.1/src'
make  check-am
make[2]: Wejście do katalogu '/home/adrian/Downloads/protobuf-3.4.1/src'
make  protoc protobuf-test protobuf-lazy-descriptor-test protobuf-lite-test test_plugin protobuf-lite-arena-test no-warning-test zcgzip zcgunzip
make[3]: Wejście do katalogu '/home/adrian/Downloads/protobuf-3.4.1/src'
make[3]: 'protoc' jest aktualne.
make[3]: 'protobuf-test' jest aktualne.
make[3]: 'protobuf-lazy-descriptor-test' jest aktualne.
make[3]: 'protobuf-lite-test' jest aktualne.
make[3]: 'test_plugin' jest aktualne.
make[3]: 'protobuf-lite-arena-test' jest aktualne.
g++ -DHAVE_CONFIG_H -I. -I..    -pthread -DHAVE_PTHREAD=1 -DHAVE_ZLIB=1 -Wall -Wextra -Werror -Wno-unused-parameter -g -DNDEBUG -MT google/protobuf/no_warning_test-map_lite_unittest.pb.o -MD -MP -MF google/protobuf/.deps/no_warning_test-map_lite_unittest.pb.Tpo -c -o google/protobuf/no_warning_test-map_lite_unittest.pb.o `test -f 'google/protobuf/map_lite_unittest.pb.cc' || echo './'`google/protobuf/map_lite_unittest.pb.cc
In file included from ./google/protobuf/generated_message_table_driven.h:35:0,
                 from ./google/protobuf/map_lite_unittest.pb.h:25,
                 from google/protobuf/map_lite_unittest.pb.cc:5:
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, protobuf_unittest::ForeignMessageLite, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)11, 0>; Base = google::protobuf::MessageLite; Key = int; Value = protobuf_unittest::ForeignMessageLite; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)11; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, int, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)5, 0>; Base = google::protobuf::MessageLite; Key = int; Value = int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<long int, long int, (google::protobuf::internal::WireFormatLite::FieldType)3, (google::protobuf::internal::WireFormatLite::FieldType)3, 0>; Base = google::protobuf::MessageLite; Key = long int; Value = long int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)3; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)3; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<unsigned int, unsigned int, (google::protobuf::internal::WireFormatLite::FieldType)13, (google::protobuf::internal::WireFormatLite::FieldType)13, 0>; Base = google::protobuf::MessageLite; Key = unsigned int; Value = unsigned int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)13; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)13; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<long unsigned int, long unsigned int, (google::protobuf::internal::WireFormatLite::FieldType)4, (google::protobuf::internal::WireFormatLite::FieldType)4, 0>; Base = google::protobuf::MessageLite; Key = long unsigned int; Value = long unsigned int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)4; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)4; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, int, (google::protobuf::internal::WireFormatLite::FieldType)17, (google::protobuf::internal::WireFormatLite::FieldType)17, 0>; Base = google::protobuf::MessageLite; Key = int; Value = int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)17; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)17; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<long int, long int, (google::protobuf::internal::WireFormatLite::FieldType)18, (google::protobuf::internal::WireFormatLite::FieldType)18, 0>; Base = google::protobuf::MessageLite; Key = long int; Value = long int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)18; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)18; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<unsigned int, unsigned int, (google::protobuf::internal::WireFormatLite::FieldType)7, (google::protobuf::internal::WireFormatLite::FieldType)7, 0>; Base = google::protobuf::MessageLite; Key = unsigned int; Value = unsigned int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)7; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)7; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<long unsigned int, long unsigned int, (google::protobuf::internal::WireFormatLite::FieldType)6, (google::protobuf::internal::WireFormatLite::FieldType)6, 0>; Base = google::protobuf::MessageLite; Key = long unsigned int; Value = long unsigned int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)6; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)6; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, int, (google::protobuf::internal::WireFormatLite::FieldType)15, (google::protobuf::internal::WireFormatLite::FieldType)15, 0>; Base = google::protobuf::MessageLite; Key = int; Value = int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)15; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)15; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<long int, long int, (google::protobuf::internal::WireFormatLite::FieldType)16, (google::protobuf::internal::WireFormatLite::FieldType)16, 0>; Base = google::protobuf::MessageLite; Key = long int; Value = long int; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)16; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)16; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, float, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)2, 0>; Base = google::protobuf::MessageLite; Key = int; Value = float; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)2; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, double, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)1, 0>; Base = google::protobuf::MessageLite; Key = int; Value = double; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)1; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<bool, bool, (google::protobuf::internal::WireFormatLite::FieldType)8, (google::protobuf::internal::WireFormatLite::FieldType)8, 0>; Base = google::protobuf::MessageLite; Key = bool; Value = bool; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)8; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)8; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, (google::protobuf::internal::WireFormatLite::FieldType)9, (google::protobuf::internal::WireFormatLite::FieldType)9, 0>; Base = google::protobuf::MessageLite; Key = std::__cxx11::basic_string<char>; Value = std::__cxx11::basic_string<char>; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)9; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)9; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, std::__cxx11::basic_string<char>, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)12, 0>; Base = google::protobuf::MessageLite; Key = int; Value = std::__cxx11::basic_string<char>; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)12; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, protobuf_unittest::MapEnumLite, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)14, 0>; Base = google::protobuf::MessageLite; Key = int; Value = protobuf_unittest::MapEnumLite; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)14; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, protobuf_unittest::ForeignMessageArenaLite, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)11, 0>; Base = google::protobuf::MessageLite; Key = int; Value = protobuf_unittest::ForeignMessageArenaLite; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)11; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, protobuf_unittest_no_arena::ForeignMessageLite, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)11, 0>; Base = google::protobuf::MessageLite; Key = int; Value = protobuf_unittest_no_arena::ForeignMessageLite; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)11; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, protobuf_unittest::TestRequiredLite, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)11, 0>; Base = google::protobuf::MessageLite; Key = int; Value = protobuf_unittest::TestRequiredLite; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)11; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, protobuf_unittest::Proto2MapEnumLite, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)14, 0>; Base = google::protobuf::MessageLite; Key = int; Value = protobuf_unittest::Proto2MapEnumLite; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)14; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, protobuf_unittest::Proto2MapEnumPlusExtraLite, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)14, 0>; Base = google::protobuf::MessageLite; Key = int; Value = protobuf_unittest::Proto2MapEnumPlusExtraLite; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)14; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
./google/protobuf/map_entry_lite.h: In member function ‘bool google::protobuf::internal::MapEntryImpl<Derived, Base, Key, Value, kKeyFieldType, kValueFieldType, default_enum_value>::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*) [with Derived = google::protobuf::internal::MapEntryLite<int, protobuf_unittest::TestAllTypesLite, (google::protobuf::internal::WireFormatLite::FieldType)5, (google::protobuf::internal::WireFormatLite::FieldType)11, 0>; Base = google::protobuf::MessageLite; Key = int; Value = protobuf_unittest::TestAllTypesLite; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)5; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)11; int default_enum_value = 0]’:
./google/protobuf/map_entry_lite.h:205:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (!input->ExpectTag(kValueTag)) break;
           ^~
./google/protobuf/map_entry_lite.h:208:9: note: here
         case kValueTag:
         ^~~~
cc1plus: all warnings being treated as errors
make[3]: *** [Makefile:4114: google/protobuf/no_warning_test-map_lite_unittest.pb.o] Błąd 1
make[3]: Opuszczenie katalogu '/home/adrian/Downloads/protobuf-3.4.1/src'
make[2]: *** [Makefile:7970: check-am] Błąd 2
make[2]: Opuszczenie katalogu '/home/adrian/Downloads/protobuf-3.4.1/src'
make[1]: *** [Makefile:7973: check] Błąd 2
make[1]: Opuszczenie katalogu '/home/adrian/Downloads/protobuf-3.4.1/src'
make: *** [Makefile:1480: check-recursive] Błąd 1
```

--- a/src/google/protobuf/stubs/port.h
+++ b/src/google/protobuf/stubs/port.h
@@ -243,6 +243,8 @@ static const uint64 kuint64max = GOOGLE_ULONGLONG(0xFFFFFFFFFFFFFFFF);
      __has_cpp_attribute(clang::fallthrough)
 #  define GOOGLE_FALLTHROUGH_INTENDED [[clang::fallthrough]]
 # endif
+#elif defined(__GNUC__) && __GNUC__ > 6
+# define GOOGLE_FALLTHROUGH_INTENDED [[gnu::fallthrough]]
 #endif
 
 #ifndef GOOGLE_FALLTHROUGH_INTENDED

